src/Relay/Connection/Paginator.php

```diff
@@ -4,45 +4,47 @@
 
 namespace Overblog\GraphQLBundle\Relay\Connection;
 
+use GraphQL\Executor\Promise\Promise;
 use Overblog\GraphQLBundle\Definition\ArgumentInterface;
+use function call_user_func;
+use function call_user_func_array;
+use function count;
+use function is_callable;
+use function is_numeric;
+use function max;
 
 class Paginator
 {
     public const MODE_REGULAR = false;
     public const MODE_PROMISE = true;
 
+    private bool $promise;
+    private int $totalCount;
+    private ConnectionBuilder $connectionBuilder;
+
     /** @var callable */
     private $fetcher;
 
-    /** @var bool */
-    private $promise;
-
-    /** @var int */
-    private $totalCount;
-
-    /** @var ConnectionBuilder */
-    private $connectionBuilder;
-
     public function __construct(callable $fetcher, bool $promise = self::MODE_REGULAR, ConnectionBuilder $connectionBuilder = null)
     {
         $this->fetcher = $fetcher;
         $this->promise = $promise;
-        $this->connectionBuilder = $connectionBuilder ?: new ConnectionBuilder();
+        $this->connectionBuilder = $connectionBuilder ?? new ConnectionBuilder();
     }
 
     /**
      * @param int|callable $total
      *
-     * @return ConnectionInterface|object A connection or a promise
```

Comment:
could we keep `ConnectionInterface|object` object represent promise.

Reply:
Tested this method with Xdebug and fixed to `Connection|Promise`